src/plugins/python/python.cpp

```diff
@@ -0,0 +1,401 @@
+		Py_DECREF(data->instance);
+		data->instance = NULL;
+	}
+	PyGILState_Release(gstate);
+
+	/* destroy python if plugin isn't used anymore */
+	//FIXME python reinitialization is known to be buggy
+	pthread_mutex_lock(&mutex);
+	open_cnt--;
+	if (!open_cnt && Py_IsInitialized())
+		Py_Finalize();
+	pthread_mutex_unlock(&mutex);
+
+	delete data;
+	return ret;
+}
+
+int ELEKTRA_PLUGIN_FUNCTION(Python, Get)(ckdb::Plugin *handle, ckdb::KeySet *returned,
+	ckdb::Key *parentKey)
+{
+	moduleData *data = static_cast<moduleData *>(elektraPluginGetData(handle));
+	if (data != NULL)
+	{
+		/* call python function */
+		int ret = 0;
+		PyGILState_STATE gstate = PyGILState_Ensure();
+		PyObject *func = PyObject_GetAttrString(data->instance, "get");
+		if (func)
+		{
+			PyObject *arg0 = Python_fromSWIG(returned);
+			PyObject *arg1 = Python_fromSWIG(parentKey);
+			PyObject *args = Py_BuildValue("(OO)", arg0, arg1);
+			ret = Python_CallFunction_Int(func, args);
+			Py_DECREF(arg0);
+			Py_DECREF(arg1);
+			Py_DECREF(args);
+			Py_DECREF(func);
+		}
+		PyGILState_Release(gstate);
+		return ret;
+	}
+
+#define _MODULE_CONFIG_PATH "system/elektra/modules/" ELEKTRA_PLUGIN_NAME
+	if (!strcmp(keyName(parentKey), _MODULE_CONFIG_PATH))
+	{
+		KeySet *n;
+		ksAppend(returned, n = ksNew(30,
+			keyNew(_MODULE_CONFIG_PATH,
+				KEY_VALUE, "python interpreter waits for your orders", KEY_END),
+			keyNew(_MODULE_CONFIG_PATH "/exports", KEY_END),
+			keyNew(_MODULE_CONFIG_PATH "/exports/get",
+				KEY_FUNC, ELEKTRA_PLUGIN_FUNCTION(Python, Get),
+				KEY_END),
+			keyNew(_MODULE_CONFIG_PATH "/exports/set",
+				KEY_FUNC, ELEKTRA_PLUGIN_FUNCTION(Python, Set),
+				KEY_END),
+			keyNew(_MODULE_CONFIG_PATH "/exports/error",
+				KEY_FUNC, ELEKTRA_PLUGIN_FUNCTION(Python, Error),
+				KEY_END),
+#include "readme_python.c"
+			keyNew(_MODULE_CONFIG_PATH "/infos/version",
+				KEY_VALUE, PLUGINVERSION, KEY_END),
+			KS_END));
+		ksDel(n);
+		return 1;
+	}
+	return 1;
+}
+
+int ELEKTRA_PLUGIN_FUNCTION(Python, Set)(ckdb::Plugin *handle, ckdb::KeySet *returned,
+	ckdb::Key *parentKey)
+{
+	int ret = 0;
+	moduleData *data = static_cast<moduleData *>(elektraPluginGetData(handle));
+	if (data != NULL)
+	{
+		/* call python function */
+		PyGILState_STATE gstate = PyGILState_Ensure();
+		PyObject *func = PyObject_GetAttrString(data->instance, "set");
+		if (func)
+		{
+			PyObject *arg0 = Python_fromSWIG(returned);
+			PyObject *arg1 = Python_fromSWIG(parentKey);
+			PyObject *args = Py_BuildValue("(OO)", arg0, arg1);
+			ret = Python_CallFunction_Int(func, args);
+			Py_DECREF(arg0);
+			Py_DECREF(arg1);
+			Py_DECREF(args);
+			Py_DECREF(func);
+		}
+		PyGILState_Release(gstate);
+	}
+	return ret;
+}
+
+int ELEKTRA_PLUGIN_FUNCTION(Python, Error)(ckdb::Plugin *handle, ckdb::KeySet *returned,
+	ckdb::Key *parentKey)
+{
+	int ret = 0;
+	moduleData *data = static_cast<moduleData *>(elektraPluginGetData(handle));
+	if (data != NULL)
+	{
+		/* call python function */
```

Comment:
calling python functions looks identical, why not wrap this in a function for Get, Set, Error?